src/yb/master/master-path-handlers.cc

```diff
@@ -1428,6 +1430,25 @@ void MasterPathHandlers::HandleGetClusterConfig(
   << "<pre class=\"prettyprint\">" << config.DebugString() << "</pre>";
 }
 
+void MasterPathHandlers::HandleGetClusterConfigJSON(
+  const Webserver::WebRequest& req, Webserver::WebResponse* resp) {
+  std::string jsonOutput;
+  std::stringstream *output = &resp->output;
+  master_->catalog_manager()->AssertLeaderLockAcquiredForReading();
+
+  SysClusterConfigEntryPB config;
+  Status s = master_->catalog_manager()->GetClusterConfig(&config);
+  if (!s.ok()) {
+    *output << "<div class=\"alert alert-warning\">" << s.ToString() << "</div>";
```

Comment:
The error should be reported as part of the JSON object instead of as a HTML string. See HandleHealthCheck for example.

Reply:
Done